src/qmapshack/gis/rte/router/CRouterRoutino.cpp

```diff
@@ -248,14 +242,47 @@ void CRouterRoutino::buildDatabaseList()
                 continue;
             }
 
+            // qDebug() << "buildDatabase" << prefix;
+
 #ifdef Q_OS_WIN
             Routino_Database * data = Routino_LoadDatabase(dir.absolutePath().toLocal8Bit(), prefix.toLocal8Bit());
 #else
             Routino_Database * data = Routino_LoadDatabase(dir.absolutePath().toUtf8(), prefix.toUtf8());
 #endif
+            QVariantMap dmap;
+
             if(data)
             {
-                comboDatabase->addItem(prefix.replace("_", " "), quint64(data));
+                /* determine the profile to use */
+                dmap["db"] = QVariant ((qulonglong)data);
+
+                IAppSetup *setup = IAppSetup::getPlatformInstance();
+
+                /* check possible profiles.xml locations and keep the first active one. */
+                dmap["prefixpath"] = "";
+                QStringList profilespaths = {
+                    dir.filePath(prefix+"-profiles.xml"),
+                    dir.filePath("profiles.xml"),
+                    setup->routinoPath("profiles.xml").toUtf8()
+                };
+                QStringListIterator ppit(profilespaths);
+                while( ppit.hasNext() )
+                {
+                    QFileInfo pinfo = QFileInfo(ppit.next());
+                    if( pinfo.isReadable())
+                    {
+                        dmap["prefixpath"] = pinfo.filePath();
+                        break;
+                    }
+                }
+                if( dmap["prefixpath"].toString().length() == 0)
+                {
+                    QMessageBox::critical(this, "Routino...", xlateRoutinoError(Routino_errno), QMessageBox::Abort);
```

Comment:
Where does `Routino_errno` come from? It's not set by any Routino API call. You are just gathering a profile path. 
Why are you using `QMessageBox::Abort` but you do not abort. In fact you add the item even with an invalid path.

Reply:
Sorry too much cut and past.  Defined a new Error message as a critical warning that routino will be non functional.